### ISSUE TITLE
Implement missing members suggestion shadows type parameters

### ISSUE THREAD

**Issue by thirdsgames** [rust-analyzer/rust-analyzer#6870]
In the following minimal example in latest nightly (using a `struct<T> DiagnosticResult { ... }`), the "implement missing members" intention shadows the type parameter `T`.
![image](https://user-images.githubusercontent.com/50671761/102070002-70e16180-3df6-11eb-999b-636e0bcf1c62.png)
![image](https://user-images.githubusercontent.com/50671761/102070019-75a61580-3df6-11eb-9153-234583bd8a1c.png)
This creates rather confusing error messages if you didn't actually read the code that it inserted!
![image](https://user-images.githubusercontent.com/50671761/102070153-a1c19680-3df6-11eb-93b3-9e23931ff1bc.png)
The two type parameters `T` and `T` are different, but I imagine that in a more complex scenario than this, it could hide the source of bugs. I think rust-analyzer should probably output code similar to
![image](https://user-images.githubusercontent.com/50671761/102070061-82c30480-3df6-11eb-8939-8b15df9319c5.png)

**Comment by Natural-selection1** [comment]
Yes, we did not check for possible duplicate names, but this will be fixed in https://github.com/rust-lang/rust-analyzer/pull/21118.
